### ISSUE TITLE
Make return value of saga testable when using `expectSaga`

### ISSUE THREAD

**Issue by adamterlson** [jfairbank/redux-saga-test-plan#72]
Is it possible to test the return value of a saga when using `expectSaga`?  It is of course possible when using `testSaga`, but that would require much more orchestration just to check the return value.

Is it perhaps possible to resolve the promise generated by `expectSaga(mySaga).run()` with the return value of the saga?

Here's some code similar to my desired usage if it helps:

```javascript
const fetchPagedSampleRequest = jest.fn()

function *fetchAll(channel) {
    const responseOne = fetchPagedSampleRequest()
    yield put(channel, responseOne)
    const responseTwo = fetchPagedSampleRequest()
    yield put(channel, responseTwo)
    const responseThree = fetchPagedSampleRequest()
    yield put(channel, responseThree)

    return [responseOne, responseTwo, responseThree]
}

const chan = channel()
const requestResponses = ['one', 'two', 'three']
mock(fetchPagedSampleRequest)
    .mockImplementationOnce(() => requestResponses[0])
    .mockImplementationOnce(() => requestResponses[1])
    .mockImplementationOnce(() => requestResponses[2])

it('should give a return value', function() {
        return expectSaga(fetchAll, chan)
            .put(chan, requestResponses[0])
            .put(chan, requestResponses[1])
            .put(chan, requestResponses[2])
            .run()
            .then((result) => {
                expect(result).toEqual(requestResponses) // <-- One possible solution?
            })
})

**Comment by jfairbank** [comment]
Hi, @adamterlson!

This is a great idea, but I'd rather introduce an assertion method like `.returns` over resolving the promise with the return value. I want to be careful introducing the return value as the resolution value there because it might be useful in the future to resolve with other things like all the effects that were handled over the lifetime of the saga. Using a `.returns` method is more in line with the philosophy of the chaining api too.

**Comment by adamterlson** [comment]
That is of course a much better implementation, I agree. Thank you for the response, @jfairbank.

**Comment by jfairbank** [comment]
@adamterlson This is available in v2.4.0 now via the `returns` method.

```js
function* saga() {
  return { hello: 'world' };
}

it('returns a greeting', () => {
  return expectSaga(saga)
    .returns({ hello: 'world' })
    .run();
});
```

Please let me know if you have any issues. Thanks!
